File: api-xmlsignature/csharp/test/src/dev.esya.api.xmlsignature/unit/signerHelpers/ParameterGetterFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using tr.gov.tubitak.esya.api.xmlsignature.unit.test.signerHelpers;
using tr.gov.tubitak.uekae.esya.api.xmlsignature.unit.signerHelpers;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature.unit.signerHelpers
{
    public class ParameterGetterFactory
    {
        private static BaseParameterGetter parameterGetter = null;
        public static BaseParameterGetter getParameterGetter()
        {
            if (parameterGetter == null)
            {
                parameterGetter = new UgTestParameterGetter();
            }
            return parameterGetter;
        }
    }
}

File: api-xmlsignature/testsrc/dev/esya/api/xmlsignature/legacy/signerHelpers/ParameterGetterFactory.java
package dev.esya.api.xmlsignature.legacy.signerHelpers;

/**
 * Created with IntelliJ IDEA.
 * User: yavuz.kahveci
 * Date: 30.01.2013
 * Time: 14:10
 * To change this template use File | Settings | File Templates.
 */
public class ParameterGetterFactory {

    private static BaseParameterGetter bpg = null;

    public static BaseParameterGetter getParameterGetter()
    {
        if(bpg == null)
        {
            bpg = new UgTestParameterGetter();
        }
        return bpg;
    }
}
